spec/dish_and_menu_spec.rb

```diff
@@ -0,0 +1,136 @@
+            expect(subject.shopping_menu).to eq nil
+        end
+
+        it 'Calls the shopping menu when user inputs 1' do
+            allow(subject).to receive(:shopping_options).and_return nil
+            allow(subject).to receive(:gets).and_return("1","2")
+            expect(subject).to receive(:menu_options)
+            subject.shopping_menu
+        end
+    end
+
+    describe 'shopping_options' do
+        before(:each) do
+            allow(subject).to receive(:puts).and_return nil
+        end
+
+        it 'Should display starters if user inputs 1' do
+            allow(subject).to receive(:gets).and_return("1", "4")
+            allow(subject).to receive(:checkout).and_return nil
+            expect(subject).to receive(:display_menu)
+            subject.menu_options
+        end
+
+        it 'Should display starters if user inputs 2' do
+            allow(subject).to receive(:gets).and_return("2", "4")
+            allow(subject).to receive(:checkout).and_return nil
+            expect(subject).to receive(:display_menu)
+            subject.menu_options
+        end
+
+        it 'Should display starters if user inputs 3' do
+            allow(subject).to receive(:gets).and_return("3", "4")
+            allow(subject).to receive(:checkout).and_return nil
+            expect(subject).to receive(:display_menu)
+            subject.menu_options
+        end
+
+        it 'Should call checkout if user inputs 4' do
+            allow(subject).to receive(:gets).and_return("4")
+            allow(subject).to receive(:checkout).and_return nil
+            expect(subject).to receive(:checkout)
+            subject.menu_options
+        end
+    end
+
+    describe 'checkout' do
+        let(:dish1) {double(:price => "5.00", :name => "Fish and Chips")}
+        let(:dish2) {double(:price => "2.50", :name => "Ice cream")}
+
+        before(:each) do
+            allow(subject).to receive(:puts).and_return nil
+            allow(subject).to receive(:print_receipt).and_return(nil)
+            allow(subject).to receive(:send_text_confirmation).and_return(nil)
+        end
+
+        it 'Totals basket and calls print receipt, calls the basket check' do
+            basket = [dish1, dish2]
+            allow(subject).to receive(:check_order?).and_return(true)
+            expect(subject).to receive(:print_receipt)
+            expect(subject).to receive(:send_text_confirmation)
+            expect(subject.checkout(basket)).to eq 7.5
+        end
+
+        it 'Returns nil if user cancels order' do
+            basket = [dish1, dish2]
+            allow(subject).to receive(:check_order?).and_return(false)
+            expect(subject).to receive(:print_receipt)
+            expect(subject.checkout(basket)).to eq nil
+        end
+    end
+
+    describe 'send_text_confirmation' do
+        it 'Sends text and returns true if text is sent' do
+            expect(subject.send_text_confirmation).to eq true
+        end
+    end
+
+end
```

Comment:
I like the structure of your testing, it then creates a well readable `rspec --format documentation`. 
Stubbing and doubling done clearly as well.